src/main/MainScreenContainer.js

```diff
@@ -40,11 +40,11 @@ class MainScreenContainer extends React.Component {
     }
   }
 
-  doNarrow = (newNarrow = [], anchor: number = Number.MAX_SAFE_INTEGER) => {
+  doNarrow = (newNarrow = [], anchor: number = Number.MAX_SAFE_INTEGER, message) => {
     const { auth, switchNarrow, messages } = this.props;
     registerAppActivity(auth);
     requestIdleCallback(() =>
-      switchNarrow(newNarrow, messages.filter(msg => msg.id === anchor)
+      switchNarrow(newNarrow, message || messages.filter(msg => msg.id === anchor)
```

Comment:
I am not sure about this line. We don't want to return an object || array.
Maybe your change is good in that regard, but we want to change messages.filter to `find`.